src/Microsoft.ML/Runtime/EntryPoints/MacroUtils.cs

```diff
@@ -186,13 +186,29 @@ public static TrainerKinds SignatureTypeToTrainerKind(Type sigType)
         public static TrainerKinds[] SignatureTypesToTrainerKinds(IEnumerable<Type> sigTypes) =>
             sigTypes.Select(SignatureTypeToTrainerKind).ToArray();
 
-        public static string GetTrainerName(TrainerKinds kind) => TrainerKindDict[kind].TrainerFunctionName;
+        private static string GetTrainerName(TrainerKinds kind) => TrainerKindDict[kind].TrainerFunctionName;
 
         public static T TrainerKindApiValue<T>(TrainerKinds trainerKind)
         {
             if (Enum.GetName(typeof(TrainerKinds), trainerKind) is string name)
                 return (T)Enum.Parse(typeof(T), name);
             throw new Exception($"Could not interpret enum value: {trainerKind}");
         }
+
+        public static bool IsTrainerOfKind(Type type, TrainerKinds trainerKind)
+        {
+            if (type.Name == "BinaryLogisticRegressor")
+                return trainerKind == TrainerKinds.SignatureBinaryClassifierTrainer;
+            if (type.Name == "LogisticRegressor")
+                return trainerKind == TrainerKinds.SignatureMultiClassClassifierTrainer;
```

Comment:
God, this is so ugly.
Is there a way to cast EntryPoint type to real type? Maybe you can use EntryPointCatalog to find mapping, and after you get real type you can maybe get prediction type, or get kind through reflection?
This is unmaintainable.

Reply:
A better solution might be harder to imagine though. We might let this slide for now since this change will fix things that are broken now, but we ought definitely to file a new issue so we can imagine something a bit less fragile than this system.

---
In reply to: [187421786](https://github.com/dotnet/machinelearning/pull/113#discussion_r187421786) [](ancestors = 187421786)

Reply:
Could this be addressed when we change the mechanism to use the EntryPoint attribute?

Reply:
The entry-point attribute may indeed be the best place to put a category like this -- something akin to the old-style `PredictionKind` enum, but not that please. :D :D :D

---
In reply to: [187457279](https://github.com/dotnet/machinelearning/pull/113#discussion_r187457279) [](ancestors = 187457279)